user/views.py

```diff
@@ -188,13 +229,39 @@ def add_movie(request):
 def get_item(dictionary, key):
     return dictionary.get(key)
 
+genre_colors = {
```

Comment:
Why don't you add the functionality of detecting the tag in the add_movie view and save the tag in the database. This way repeated processing can be avoided